trade_remedies_caseworker/organisations/views.py

```diff
@@ -261,6 +272,251 @@ def post(
+                "name": form.cleaned_data["contact_name"],
+                "email": form.cleaned_data["contact_email"],
+                "organisation": organisation["id"],
+            }
+        )
+        self.request.session["selected_contacts"] = [self.client.contacts(contact["id"]).id]
+        self.request.session["new_contact"] = True
+        return super().form_valid(form)
+
+    def get_next_url(self, form=None):
+        return reverse(
+            "organisations:invite-party-check",
+            kwargs={
+                "case_id": self.kwargs["case_id"],
+                "organisation_id": self.kwargs["organisation_id"],
+            },
+        )
+
+
+class OrganisationInviteReviewView(BaseOrganisationInviteView):
+    template_name = "organisations/invite_party_check.html"
+    form_class = ValidationForm
+
+    def get_selected_contacts(self):
+        selected_contacts = []
+        if "selected_contacts" in self.request.session:
+            for contact_id in self.request.session["selected_contacts"]:
+                contact = self.client.contacts(contact_id)
+                selected_contacts.append(contact)
+        return selected_contacts
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context.update(self.request.GET)
+
+        context["case_id"] = self.kwargs["case_id"]
+        context["organisation_id"] = self.kwargs["organisation_id"]
+        context["case"] = self.client.cases(self.kwargs["case_id"])
+        context["organisation"] = self.client.organisations(self.kwargs["organisation_id"])
+
+        context["selected_contacts"] = self.get_selected_contacts()
+        context["new_contact"] = self.request.session["new_contact"]
+
+        return context
+
+    def get_next_url(self, form=None):
+        return reverse(
+            "organisations:invite-party-complete",
+            kwargs={
+                "case_id": self.kwargs["case_id"],
+            },
+        )
+
+    def create_invitation(self, contact):
+        new_invitation = self.client.invitations(
+            {
+                "organisation": self.kwargs["organisation_id"],
+                "case": self.kwargs["case_id"],
+                "contact": contact.id,
+                "invalid": True,
+                "invitation_type": 3,
+            }
+        )
+        return new_invitation
+
+    def post(self, request, *args, **kwargs):
+        # create and invitation for each contact and send
+        for contact in self.get_selected_contacts():
+            invitation = self.create_invitation(contact)
+            invitation.send()
+        return redirect(self.get_next_url())
+
+
+class OrganisationInviteCompleteView(BaseOrganisationInviteView):
+    template_name = "organisations/invite_party_complete.html"
+    form_class = ValidationForm
+
+    def clean_session_data(self):
+        # clean up session data
+        try:
+            del self.request.session["new_contact"]
```

Comment:
we can use .pop() here to delete keys from a dictionary when we're not sure if they exist.

```suggestion
            self.request.session.pop("new_contact", None)
```